swupd/create_manifests.go

```diff
@@ -267,14 +267,22 @@ func processBundles(ui UpdateInfo, c config, numWorkers int) ([]*Manifest, error
 	return newManifests, nil
 }
 
-func addUnchangedManifests(appendTo *Manifest, appendFrom *Manifest) {
+func addUnchangedManifests(appendTo *Manifest, appendFrom *Manifest, bundles []string) {
 	for _, f := range appendFrom.Files {
-		if f.findFileNameInSlice(appendTo.Files) == nil {
-			if f.Name == IndexBundle {
-				// this is generated new each time
-				continue
+		if f.findFileNameInSlice(appendTo.Files) != nil {
+			continue
+		}
+
+		if f.Name == IndexBundle {
+			// this is generated new each time
+			continue
+		}
+
+		for _, bundle := range bundles {
```

Comment:
This for loop should just be 
```go
if f.findFileNameInSlice(bundle) != nil {
    appendTo.Files = append(appendTo.Files, f)
}
```
right now that function isn't optimized but it could be at a later point and this code would get the benefit.

Reply:
oh wait, that function only works for `*File` slices, not `string` slices. Never mind...